src/test/java/com/github/britter/junit/jupiter/testkit/TestkitExamplesTests.java

```diff
@@ -0,0 +1,94 @@
+    void execute_all_methods_of_class(TestExecutionStatistics statistics) {
+        assertEquals(1, statistics.getContainerStatistics().getStartedCount());
+        assertEquals(1, statistics.getContainerStatistics().getSuccessfulCount());
+        assertEquals(0, statistics.getContainerStatistics().getFailedCount());
+
+        assertEquals(3, statistics.getTestStatistics().getStartedCount());
+        assertEquals(2, statistics.getTestStatistics().getSuccessfulCount());
+        assertEquals(1, statistics.getTestStatistics().getSkippedCount());
+        assertEquals(0, statistics.getTestStatistics().getFailedCount());
+    }
+
+    @ExtensionTest(InnerTestCase.class)
+    void execute_all_methods_of_class_only_analysing_only_container_stats(@ContainerStatistics Statistics containerStats) {
+        assertEquals(1, containerStats.getStartedCount());
+        assertEquals(1, containerStats.getSuccessfulCount());
+        assertEquals(0, containerStats.getFailedCount());
+    }
+
+    @ExtensionTest(InnerTestCase.class)
+    void execute_all_methods_of_class_only_analysing_only_test_stats(@TestStatistics Statistics testStats) {
+        assertEquals(3, testStats.getStartedCount());
+        assertEquals(2, testStats.getSuccessfulCount());
+        assertEquals(1, testStats.getSkippedCount());
+        assertEquals(0, testStats.getFailedCount());
+    }
+
+    @ExtensionTest(value = InnerTestCase.class, methods = "test_method_1")
+    void execute_selected_methods_of_class(@TestStatistics Statistics testStats) {
+        assertEquals(1, testStats.getStartedCount());
+        assertEquals(1, testStats.getSuccessfulCount());
+        assertEquals(0, testStats.getFailedCount());
+    }
+
+    @ExtensionTest(value = InnerTestCase.class, methods = "test_method_2")
+    void execute_selected_methods_of_class_analysing_event(ExecutionEvents events) {
+        assertEquals(5, events.asList().size());
+    }
+
+    @ExtensionTest(value = InnerTestCase.class, methods = "test_method_2")
+    void execute_selected_methods_of_class_filtering_only_skipped_events(@EventsFilter(SKIPPED) List<ExecutionEvent> events) {
+        assertEquals(1, events.size());
+    }
+
+    @ExtensionTest(value = InnerTestCase.class, methods = "test_method_2")
+    void execute_selected_methods_of_class_filtering_only_container_events(@EventsFilter(type = CONTAINER) List<ExecutionEvent> events) {
+        assertEquals(1, events.size());
+    }
+
+    @ExtensionTest(value = InnerTestCase.class, methods = "test_method_2")
+    void execute_selected_methods_of_class_filtering_only_failed_tests(@EventsFilter(status = FAILED) List<ExecutionEvent> events) {
+        assertEquals(0, events.size());
+    }
+
+    @ExtensionTest(value = InnerTestCase.class, methods = "test_method_2")
+    void execute_selected_methods_of_class_with_complex_filtering(@EventsFilter(value = ExecutionEvent.Type.FINISHED, type = TEST, status = FAILED) List<ExecutionEvent> events) {
```

Comment:
Pretty neat, is the plan here to just auto-capture the events using `@ExtensionTest` and inject based on the parameters `@EventsFilter` etc?

Reply:
I like the idea of specifying what you're interested in. But I don't know whether this annotation based filter API is good for evolving the API.